src/setup/rgw.py

```diff
@@ -458,8 +443,8 @@ def _generate_ssl_cert(conf: MappedConf):
                 raise SetupError(errno.EINVAL, f'Failed to generate self signed ssl certificate: {e}')
 
     @staticmethod
-    def _update_hax_endpoint_and_create_admin(conf: MappedConf, data_pod_hostname: str):
-        """Update motr_ha(hax) endpoint values to rgw config file."""
+    def _update_hax_endpoint(conf: MappedConf, data_pod_hostname: str):
```

Comment:
data_pod_hostname is not being validated at all, like it should not be a null value or empty sting, in which case it should return an non-zero error code and consuming function should skip that data_pod_hostname

Reply:
@selvakumaar5496 please address above comment for error handling separately. rest of the code looks fine, merging now, so we can trigger the build, to use for other imp tasks